presto-main/src/main/java/com/facebook/presto/tdigest/TDigest.java

```diff
@@ -159,6 +159,26 @@ public static TDigest createTDigest(double compression)
         return new TDigest(compression);
     }
 
+    public static TDigest createTDigest(
+            double[] centroidMeans,
+            double[] centroidWeights,
+            double compression,
+            double min,
+            double max,
+            double sum,
+            int count)
+    {
+        TDigest tDigest = new TDigest(compression);
+        tDigest.setMinMax(min, max);
+        tDigest.setSum(sum);
+        tDigest.totalWeight = sum; // TODO: what should totalWeight be?
```

Comment:
The total weight should be the sum of the `centroidWeights` (see the sanity check at the end of `merge`)